besskge/dataset.py

```diff
@@ -284,11 +284,71 @@ def build_yago310(cls, root: Path) -> "KGDataset":
             neg_tails=None,
         )
 
+    @classmethod
+    def from_triples(
+        cls,
+        data: NDArray[np.int32],
+        split: Tuple[float, float, float] = (0.7, 0.15, 0.15),
+        entity_dict: Optional[List[str]] = None,
+        relation_dict: Optional[List[str]] = None,
+        type_offsets: Optional[Dict[str, int]] = None,
+        neg_heads: Optional[Dict[str, NDArray[np.int32]]] = None,
+        neg_tails: Optional[Dict[str, NDArray[np.int32]]] = None,
```

Comment:
`neg_heads` and `neg_tails`, in dictionary form, have the same structure of the final `triples`, i.e. each entry of the dictionary corresponds to a part of the dataset splitting (with usually only "valid"/"test" indices used, see for instance the `build_biokg` method).
Since here a train/valid/test split is not known before calling the method, I am not sure it makes much sense to have them in this form as inputs. I see two alternatives:
- have them as np.int32 array inputs, like `data`, and then construct the dictionary accordingly based on the split you generate. This however requires that the negative heads/tails are specified for all triples, including the training ones, which is usually not the case;
- remove these inputs and always set `neg_heads, neg_tails` to `None` in the returned `KGDataset` instance. This would be a "pure" from_triples method, with no negative entities specified - but maybe a bit too restrictive?

Reply:
Good point! I'm leaning towards the second option, in case users want specific negatives validation/test they can still manually instantiate the class. I might add a comment on that